### PR TITLE
infoschema: add metrics_tables that contain all metrics tables definition

### PR DESCRIPTION
**Pull request by crazycs520**
Signed-off-by: crazycs <crazycs520@gmail.com>

### What problem does this PR solve? 
* Add `information_schema.metrics_tables` that contain all metrics tables definition.
* Add `comment` column in all metrics tables to explain the metric meaning.
```sql
> select * from `METRICS_TABLES` limit 2\G
***************************[ 1. row ]***************************
METRIC_NAME     | abnormal_stores
PROMQL          | sum(pd_cluster_status{ type=~"store_disconnected_count|store_unhealth_count|store_low_space_count|store_down_count|store_offline_count|store_tombstone_count"})
GENERATE_PROMQL | sum(pd_cluster_status{ type=~"store_disconnected_count|store_unhealth_count|store_low_space_count|store_down_count|store_offline_count|store_tombstone_count"})
LABEL           | instance,type
QUANTILE        | 0.0
COMMENT         |
***************************[ 2. row ]***************************
METRIC_NAME     | connection_count
PROMQL          | tidb_server_connections{$LABEL_CONDITIONS}
GENERATE_PROMQL | tidb_server_connections{}
LABEL           | instance
QUANTILE        | 0.0
COMMENT         | TiDB current connection counts

> select `METRIC_NAME`, `AVG_VALUE`,`COMMENT` from `METRICS_SUMMARY_BY_LABEL` limit 3;
+-------------------------------+-------------------+-----------------------------------------------------------------------+
| METRIC_NAME                   | AVG_VALUE         | COMMENT                                                               |
+-------------------------------+-------------------+-----------------------------------------------------------------------+
| etcd_disk_wal_fsync_rate      | 120.598140046     | The rate of writing WAL into the persistent storage                   |
| etcd_wal_fsync_duration_0.999 |   0.0202235687646 | The quantile time consumed of writing WAL into the persistent storage |
| go_gc_count                   |   0.0799993200264 | The Go garbage collection counts per second                           |
+-------------------------------+-------------------+-----------------------------------------------------------------------+
```

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change

### CODE REVIEW

File: infoschema/tables.go [hunk 1]
```diff
@@ -2309,6 +2324,30 @@ func dataForTableTiFlashReplica(schemas []*model.DBInfo) [][]types.Datum {
...
+func dataForMetricTables(ctx sessionctx.Context) [][]types.Datum {
+	var rows [][]types.Datum
+	tables := make([]string, 0, len(MetricTableMap))
+	for name := range MetricTableMap {
+		tables = append(tables, name)
+	}
+	sort.Strings(tables)
+	for _, name := range tables {
+		schema := MetricTableMap[name]
+		genPromQL := schema.GenPromQL(ctx, nil, schema.Quantile)
```
Comment (lonng) [inline infoschema/tables.go]: I think it's unnecessary to add the `GENERATE_PROMQL` column.

File: infoschema/tables.go [hunk 2]
```diff
@@ -1131,14 +1134,25 @@ var tableInspectionResultCols = []columnInfo{
 	{"DETAILS", mysql.TypeVarchar, 256, 0, nil, nil},
 }
 
+var tableMetricTablesCols = []columnInfo{
+	{"METRIC_NAME", mysql.TypeVarchar, 64, 0, nil, nil},
```
Comment (lonng) [inline infoschema/tables.go]: It's better to name this column as `TABLE_NAME`.

File: infoschema/tables.go [hunk 3]
```diff
@@ -1131,14 +1134,25 @@ var tableInspectionResultCols = []columnInfo{
...
+	{"PROMQL", mysql.TypeVarchar, 64, 0, nil, nil},
+	{"GENERATE_PROMQL", mysql.TypeVarchar, 64, 0, nil, nil},
```
Comment (lonng) [inline infoschema/tables.go]: Prefer to remove this column, which redundant with the `PROMQL + QUANTILE`.

File: infoschema/tables.go [hunk 4]
```diff
@@ -1131,14 +1134,25 @@ var tableInspectionResultCols = []columnInfo{
...
+	{"LABEL", mysql.TypeVarchar, 64, 0, nil, nil},
```
Comment (lonng) [inline infoschema/tables.go]: Some table contains multiple labels, so I think it's better to name this column `LABELS`.

File: executor/metric_reader.go [hunk 5]
```diff
@@ -134,15 +135,15 @@ func (e *MetricRetriever) genRows(value pmodel.Value, r promQLQueryRange, quanti
 		matrix := value.(pmodel.Matrix)
 		for _, m := range matrix {
 			for _, v := range m.Values {
-				record := e.genRecord(m.Metric, v, r, quantile)
+				record := e.genRecord(m.Metric, v, quantile)
```
Comment (lonng) [inline executor/metric_reader.go]: Seems the parameter `r` at L131 can be eliminated.

Comment (crazycs520) [reply executor/metric_reader.go]: great,done.

File: infoschema/tables.go (hunk 4 above)
Comment (crazycs520) [reply infoschema/tables.go]: great,done.

File: infoschema/tables.go (hunk 3 above)
Comment (crazycs520) [reply infoschema/tables.go]: great,done.

File: infoschema/tables.go (hunk 2 above)
Comment (crazycs520) [reply infoschema/tables.go]: great,done.

File: infoschema/tables.go (hunk 1 above)
Comment (crazycs520) [reply infoschema/tables.go]: great,done.

File: infoschema/tables.go [hunk 6]
```diff
@@ -2309,6 +2323,28 @@ func dataForTableTiFlashReplica(schemas []*model.DBInfo) [][]types.Datum {
 	return rows
 }
 
+// dataForTableTiFlashReplica constructs data for table tiflash replica info.
```
Comment (AilinKid) [inline infoschema/tables.go]: comment is right?

Comment (crazycs520) [reply infoschema/tables.go]: good catch, done.

File: infoschema/tables.go [hunk 7]
```diff
@@ -2334,12 +2333,10 @@ func dataForMetricTables(ctx sessionctx.Context) [][]types.Datum {
 	sort.Strings(tables)
 	for _, name := range tables {
 		schema := MetricTableMap[name]
-		genPromQL := schema.GenPromQL(ctx, nil, schema.Quantile)
 		record := types.MakeDatums(
 			name,                             // METRIC_NAME
 			schema.PromQL,                    // PROMQL
-			genPromQL,                        // GENERATE_PROMQL
-			strings.Join(schema.Labels, ","), //LABEL
+			strings.Join(schema.Labels, ","), // LABEL
```
Comment (lonng) [inline infoschema/tables.go]:
```suggestion
			strings.Join(schema.Labels, ","), // LABELS
```

**Pull request merged**